core/src/main/scala/com/banno/kafka/producer/ProducerOps.scala

```diff
@@ -40,12 +40,64 @@ case class ProducerOps[F[_], K, V](producer: ProducerApi[F, K, V]) {
   )(implicit F: Applicative[F]): F[G[RecordMetadata]] =
     records.traverse(producer.sendAsync)
 
+  /** Sends all of the records to the producer (synchronously), so the producer
+    * may batch them. After all records are sent, asynchronously waits for all
+    * acks. Returns the write metadatas, in order. This is the only batch write
+    * operation that allows the producer to perform its own batching, while
+    * semantically blocking until all writes have succeeded. It maximizes
+    * concurrency and producer batching, and also simplicity of usage. Fails if
+    * any individual send or ack fails.
+    */
+  def sendBatch[G[_]: Traverse](
```

Comment:
Is it worth requiring that `G` have a `NonEmptyTraverse`? I think that'd allow the constraint on `F` below to be a `FlatMap`.

Reply:
TIL `NonEmptyTraverse`, had no idea. That does indeed work for `sendBatch`, the problem is down below there is no `NonEmptyTraverse[Chunk]` in fs2. Maybe we could impl one, not sure what that takes.

Reply:
Or we ditch the chunk-related stuff. It seemed cool, but no idea if it's useful in practice.

Reply:
`Chunk` does have a `toNel: Option[NonEmptyList[O]]` that could be useful?

Reply:
https://github.com/Banno/kafka4s/pull/825/commits/590ffc298f4456083928bce0755b84da86504260 shows one option: a parallel series of `NonEmpty` operations.

Reply:
Or https://github.com/Banno/kafka4s/pull/825/commits/3253892e0d17a5c6cce1851fcb78dd165d533ee6 shows one way to use `Chunk` with `NonEmptyTraverse`.